GameServerLib/Content/FileEntry.cs

```diff
@@ -0,0 +1,38 @@
+using System.IO;
+
+namespace LeagueSandbox.GameServer.Content
+{
+    public class FileEntry
+    {
+        public string Name { get; }
+        public string FullPath { get; }
+
+        private byte[] _data;
+
+        public FileEntry(string name, string fullPath, Stream stream)
```

Comment:
This is still storing full content of the file into memory.
However this time you did define a class that describes an interface for a ``FileEntry``.

You should extract this interface so that you can have multiple implementations for ``FileEntry``(example .zip or file system).
And yes this would mean that your ``ContentManager`` would have to implement ``IDisposable`` to get rid of open .zip handles which should be trivial to implement(store them in list).

So what you should implement is ``FilesystemFileEntry`` which should be wraper for ``IFilerEntry`` around this class:
https://docs.microsoft.com/en-us/dotnet/api/system.io.fileinfo?view=netframework-4.7.2

For ``ZipFileEntry`` you can wrap this class(``ZipArchiveEntry``):
https://docs.microsoft.com/en-us/dotnet/standard/io/how-to-compress-and-extract-files

Both of those provide you a way to:
1. File name
2. File name relative to root path of package 
3. ``Open`` or ``OpenRead`` which returns a stream